File: lib/screens/onboarding_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../widgets/app_widgets.dart';
import '../constants/theme.dart';

class OnboardingScreen extends StatefulWidget {
  const OnboardingScreen({Key? key}) : super(key: key);

  @override
  State<OnboardingScreen> createState() => _OnboardingScreenState();
}

class _OnboardingScreenState extends State<OnboardingScreen> {
  late PageController _pageController;
  int _currentPage = 0;

  final List<Map<String, String>> pages = [
    {
      'title': 'Take Notes',
      'subtitle': 'Capture sermon insights, scriptures, and key points',
      'icon': '📝',
    },
    {
      'title': 'Reflect & Learn',
      'subtitle': 'AI-powered reflections on spiritual lessons',
      'icon': '🤔',
    },
    {
      'title': 'Take Action',
      'subtitle': 'Convert sermons into actionable daily tasks',
      'icon': '✨',
    },
    {
      'title': 'Track Growth',
      'subtitle': 'Visualize your spiritual journey and progress',
      'icon': '📈',
    },
  ];

  @override
  void initState() {
    super.initState();
    _pageController = PageController();
  }

  @override
  void dispose() {
    _pageController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
          Expanded(
            child: PageView.builder(
              controller: _pageController,
              onPageChanged: (index) {
                setState(() => _currentPage = index);
              },
              itemCount: pages.length,
              itemBuilder: (context, index) {
                final page = pages[index];
                return Padding(
                  padding: const EdgeInsets.all(32),
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      Text(page['icon']!, style: const TextStyle(fontSize: 80)),
                      const SizedBox(height: 32),
                      Text(
                        page['title']!,
                        style: Theme.of(context).textTheme.displaySmall,
                        textAlign: TextAlign.center,
                      ),
                      const SizedBox(height: 16),
                      Text(
                        page['subtitle']!,
                        style: Theme.of(context).textTheme.bodyLarge?.copyWith(
                          color: AppTheme.textMuted,
                        ),
                        textAlign: TextAlign.center,
                      ),
                    ],
                  ),
                );
              },
            ),
          ),
          Padding(
            padding: const EdgeInsets.all(24),
            child: Column(
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: List.generate(
                    pages.length,
                    (index) => Container(
                      margin: const EdgeInsets.symmetric(horizontal: 4),
                      width: 8,
                      height: 8,
                      decoration: BoxDecoration(
                        shape: BoxShape.circle,
                        color: index == _currentPage
                            ? AppTheme.primaryNavy
                            : AppTheme.dividerLight,
                      ),
                    ),
                  ),
                ),
                const SizedBox(height: 32),
                PrimaryButton(
                  label: _currentPage == pages.length - 1 ? 'Get Started' : 'Next',
                  onPressed: () {
                    if (_currentPage == pages.length - 1) {
                      context.go('/home');
                    } else {
                      _pageController.nextPage(
                        duration: const Duration(milliseconds: 300),
                        curve: Curves.easeInOut,
                      );
                    }
                  },
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

File: src/screens/OnboardingScreen.js
import React, { useState } from 'react';
import {
  View,
  Text,
  ScrollView,
  StyleSheet,
  Animated,
  Dimensions,
} from 'react-native';
import { colors, spacing, typography, borderRadius, shadows } from '../theme';
import { Button } from './ui';

const { width } = Dimensions.get('window');

const OnboardingScreen = ({ navigation }) => {
  const [activeSlide, setActiveSlide] = useState(0);
  const scrollViewRef = React.useRef(null);

  const slides = [
    {
      icon: '📖',
      title: 'Take Notes',
      description: 'Capture sermon insights with our rich note editor',
    },
    {
      icon: '💭',
      title: 'Reflect',
      description: 'Understand lessons with AI-powered reflections',
    },
    {
      icon: '⚡',
      title: 'Apply',
      description: 'Create actionable steps for your spiritual journey',
    },
    {
      icon: '📈',
      title: 'Grow',
      description: 'Track your progress and celebrate your faith growth',
    },
  ];

  const handleScroll = (e) => {
    const slide = Math.round(e.nativeEvent.contentOffset.x / width);
    setActiveSlide(slide);
  };

  return (
    <View style={styles.container}>
      <ScrollView
        ref={scrollViewRef}
        horizontal
        pagingEnabled
        showsHorizontalScrollIndicator={false}
        onScroll={handleScroll}
        scrollEventThrottle={32}
      >
        {slides.map((slide, idx) => (
          <View key={idx} style={styles.slide}>
            <View style={styles.illustration}>
              <Text style={styles.icon}>{slide.icon}</Text>
            </View>
            <Text style={styles.title}>{slide.title}</Text>
            <Text style={styles.description}>{slide.description}</Text>
          </View>
        ))}
      </ScrollView>

      <View style={styles.bottomSection}>
        <View style={styles.dots}>
          {slides.map((_, idx) => (
            <View
              key={idx}
              style={[
                styles.dot,
                idx === activeSlide && styles.dotActive,
              ]}
            />
          ))}
        </View>

        <Button
          label={activeSlide === slides.length - 1 ? 'Get Started' : 'Next'}
          variant="primary"
          onPress={() => {
            if (activeSlide === slides.length - 1) {
              navigation.replace('MainApp');
            } else {
              scrollViewRef.current?.scrollTo({
                x: (activeSlide + 1) * width,
                animated: true,
              });
            }
          }}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background,
  },
  slide: {
    width,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: spacing.lg,
  },
  illustration: {
    width: 120,
    height: 120,
    borderRadius: 60,
    backgroundColor: colors.accent,
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: spacing.xl,
    ...shadows.lg,
  },
  icon: {
    fontSize: 56,
  },
  title: {
    ...typography.h2,
    color: colors.text,
    marginBottom: spacing.md,
    textAlign: 'center',
  },
  description: {
    ...typography.body,
    color: colors.textLight,
    textAlign: 'center',
    lineHeight: 24,
  },
  bottomSection: {
    paddingHorizontal: spacing.lg,
    paddingBottom: spacing.xl,
    paddingTop: spacing.md,
  },
  dots: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: spacing.lg,
  },
  dot: {
    width: 8,
    height: 8,
    borderRadius: 4,
    backgroundColor: colors.border,
    marginHorizontal: spacing.sm,
  },
  dotActive: {
    backgroundColor: colors.secondary,
    width: 24,
  },
});

export default OnboardingScreen;
